File: Grade 11-12/linkedList/Stack.java
package linkedList;

import java.util.Scanner;

public class Stack
{
	int size; 
	Node top;
	
	
	public void StackRun()	//For Menu-Driven Usage
	{
		int choice = 0;
		int val;
		Scanner s = new Scanner(System.in);
		
		do
		{
			System.out.println("1 - Push");
			System.out.println("2 - Pop");
			System.out.println("3 - Display Stack");
			System.out.println("9 - Exit");
			choice = s.nextInt();
			
			switch(choice)
			{
				case 1:		System.out.println("Enter value");
							val = s.nextInt();
							insertAtStart(val);
							break;
							
				case 2:		System.out.println(top.getData());
							deleteFromStart();
							break;
							
				case 3:		display();
							break;
							
				case 9:		System.exit(0);
				
				default:	System.out.println("\nEnter valid Choice");	
			}
		}while(choice != 9);
		s.close();
	}

	public void insertAtStart(int val)	//Function to add new Node at top of Stack
	{
		Node nptr = new Node(val, null);	//new Node
		size++;
		
			nptr.setLink(top);	//Attaches Node to List
			top = nptr;
	}
	
	public void deleteFromStart()	//Function to remove Node from Stack
	{
		if(size == 0)
		{
			System.out.println("Stack Empty");
			return;
		}
		
		top = top.getLink();	//sets top to next Node
		size--;
	}
	
	public void display()	//Function to display Stack
	{
		if(size == 0)
		{
			System.out.println("Empty Stack");
			return;
		}
		else
		{
			System.out.println("\n\n\n");
			Node pointer = top;
			for(int i = 0; i < size; i++)	//Traversal
			{
				System.out.print(pointer.getData() + "\t");
				pointer = pointer.getLink();
			}
			System.out.println();
		}
	}
	
}

File: VIT/CSE2003 - Data Structures and Algorithms/Stack.cpp
#include<iostream>
using namespace std;

int MAXSIZE = 10;
int TOP = -1;

bool isEmpty()
{
	if(TOP == -1)
		return true;
	return false;
}

bool isFull()
{
	if(TOP == MAXSIZE)
		return true;
	return false;
}

void push(string l[])
{
	if(isFull())
		cout << "Stack is Full. Cannot Push.\n";
	else
	{
		cout << "Enter a string\n";
		string a;
		cin >> a;

		l[++TOP] = a;
	}
}

void pop(string l[])
{
	if(isEmpty())
		cout << "Stack is Empty. Cannot Pop.\n";
	else
		TOP--;
}

string top(string l[])
{
	if(top == -1)
		return "";
	else
		return l[top];
}

void print(string l[])
{
	if(isEmpty())
		cout << "Stack is Empty" << endl;
	else
		for(int i = 0; i <= TOP; i++)
			cout << l[i] << endl;

}

int main()
{
	string List[MAXSIZE]; //List array

	    int choice;

	    do
	    {
	        cout << "\n\n1 - IsEmpty\n";
	        cout << "2 - IsFull\n";
	        cout << "3 - Push\n";
	        cout << "4 - Pop";
	        cout << "5 - Print\n";
	        cout << "6 - Exit\n\n";

	        cin >> choice;
	        switch(choice)
	        {
	            case 1:
	                cout << isEmpty() << endl << endl;
	                break;

	            case 2:
	                cout << isFull() << endl << endl;
	                break;

	            case 3: push(List);
	                    break;

	            case 4: pop(List);
	            		break;

	            case 5: print(List); break;

	            case 6: return 0;

	            default: cout << "Enter a valid choice.\n\n";
	        }
	    }
	    while(choice > 0 && choice < 7);

}
